org.eclipse.mylyn.monitor.ui/src/org/eclipse/mylyn/monitor/ui/AbstractUserInteractionMonitor.java

```diff
@@ -58,20 +54,7 @@ public void dispose() {
 	}
 
 	public void selectionChanged(IWorkbenchPart part, ISelection selection) {
-		if (part.getSite() != null && part.getSite().getWorkbenchWindow() != null) {
-			IWorkbenchWindow window = part.getSite().getWorkbenchWindow();
-			if (window instanceof IMonitoredWindow && !((IMonitoredWindow) window).isMonitored()) {
-				return;
-			}
-		}
-		if (selection == null || selection.isEmpty()) {
-			return;
-		}
-		if (!ContextCore.getContextManager().isContextActive()) {
-			handleWorkbenchPartSelection(part, selection, false);
-		} else {
-			handleWorkbenchPartSelection(part, selection, true);
-		}
+		throw new UnsupportedOperationException("should be handeled in AbstractContextInteractionMonitor");
```

Comment:
This could be very surprising implementation change

Reply:
With next commit we only report this as a Warning in the Log.